apps/workflows-dashboard/src/common/hocs/withSessionProtected/withSessionProtected.tsx

```diff
@@ -0,0 +1,35 @@
+import { setRefererUrl } from '@app/common/hocs/withSessionProtected/utils/set-referer-url';
+import { useSession } from '@app/common/hooks/useSession';
+import { LoadingSpinner } from '@app/components/atoms/LoadingSpinner';
+import { useLayoutEffect } from 'react';
+import { Navigate } from 'react-router-dom';
+
+export function withSessionProtected<TComponentProps extends object>(
+  Component: React.ComponentType<TComponentProps>,
+  signinPath = '/auth/signin',
+): React.ComponentType<TComponentProps> {
+  function Wrapper(props: TComponentProps) {
+    const { isAuthenticated, isLoading } = useSession();
+
+    useLayoutEffect(() => {
+      if (!isLoading && !isAuthenticated && location.pathname !== signinPath) {
+        setRefererUrl(location.pathname);
```

Comment:
Thoughts about handling refer URL in search query params? (it may get cleared I know)